code/__HELPERS/game.dm

```diff
@@ -262,6 +262,7 @@
 		var/atom/A = processing_list[1]
 		if(A.flags_1 & HEAR_1)
 			. += A
+			SEND_SIGNAL(A, COMSIG_ATOM_HEARER_IN_VIEW, processing_list, .)

```

Comment:
A little concerned about possible performance drawbacks here, but this is so far the easiest and least hacky solution I had.

Reply:
Don't worry about it, signals cost less than proc overhead with no listeners